### DISCUSSION TITLE
How to Remove Horizontal Scroll on a Q-Table Row? Can I make one of the columns "ellipsis" when it's too long?

### DISCUSSION THREAD

**Discussion by BenJackGill** [quasarframework/quasar#15688] [General - Components / Directives / etc]
I have a `q-table` and one of the rows is longer than desired, causing there to be a horizontal scroll.

The first column is the problem because it has a long url. So I thought I could reduce the width of that column by setting the `ellipsis` class on the first `url` column.

But doing that has no effect. The `url` column does not get truncated with an ellipsis.

**Here is my code:**

```Vue
<q-table
  flat
  :title="check.tableTitle"
  :rows="check.tableRows"
  :columns="check.tableColumns"
  :hide-header="check.hideTableTitle"
  class="full-width"
  row-key="name"
>
  <template #top>
    <div class="flex row justify-between full-width">
      <div class="text-h5">{{ check.tableTitle }}</div>
    </div>
  </template>
  <template #body-cell-url="cellProps">
    <q-td :props="cellProps">
      <a :href="cellProps.value" target="_blank">{{
        cellProps.value
      }}</a>
    </q-td>
  </template>
</q-table>


...


const tableColumns = [
  {
    name: "url",
    label: "URL",
    field: (row) => row.url,
    classes: "ellipsis", // <-- This does not work
    align: "left",
    sortable: true,
  },
  {
    name: "reason",
    label: "Reason",
    field: (row) => {
      if (row.reason === "http_header") {
        return "HTTP Header";
      } else if (row.reason === "meta_tag") {
        return "Meta Tag";
      } else if (row.reason === "robots_txt") {
        return "Robots.txt";
      } else {
        return row.reason;
      }
    },
    align: "left",
    sortable: true,
  },
];
```

**And here is a screenshot:**

<img width="817" alt="Screenshot 2023-04-07 at 5 13 03 pm" src="https://user-images.githubusercontent.com/2043257/230592680-5fe1da3e-e576-430e-bbe0-88cb436e9f40.png">

**UPDATE:**

I created a codepen here: https://codepen.io/thecashbag/pen/vYVOJRY

**Comment by MilosPaunovic** [comment]
You need to specify width property for ellipsis to have effect.

**Reply by BenJackGill** [reply]
Hmmm I still can't get it to work.

The problem is that the "url" column needs to be full-width and only use ellipsis when it gets so large that horizontal scroll happens. I can't set a fixed width on that column, it should adjust with container size.

I want both the "url" and "reason" columns to display full width and no horizontal scroll. If the columns are too wide the "url" can shrink down using the ellipsis to truncate the text thus shrinking that column.

I tried using `style: "width: 100%"` and `classes: "ellipsis"` but that didn't change anything:

```
          {
            name: "url",
            label: "URL",
            field: (row) => row.url,
            align: "left",
            classes: "ellipsis",
            style: "width: 100%",
            sortable: true,
          },
```

**Reply by BenJackGill** [reply]
I created a codepen here to see the problem: https://codepen.io/thecashbag/pen/vYVOJRY

**Comment by pdanpdan** [comment]
Make a codepen

**Reply by BenJackGill** [reply]
Created one here: https://codepen.io/thecashbag/pen/vYVOJRY

**Reply by pdanpdan** [reply]
First step is to use `table-layout: fixed`
https://codepen.io/pdanpdan/pen/GRYJOYe

But to define the column width is not so simple - use can use classes on cells in columns, or css using `:nth-child(...)`

**Reply by BenJackGill** [reply]
In this example, why does `auto-width` on the `q-td` have no effect? https://codepen.io/thecashbag/pen/LYgVvow

I thought it would shrink down the "reason" column allowing the "url" column to grow.

**Reply by pdanpdan** [reply]
Because the table tries to make the columns equal.
Unfortunately you would need some colgroup/col definitions in qtable and there is no slot for them.

Maybe you can create an issue for a slot for colgroups

**Reply by BenJackGill** [reply]
Ok I have added a feature request [here](https://github.com/quasarframework/quasar/discussions/15734).

Upvote if you also want the feature.
